src/common/ownsql.cpp

```diff
@@ -272,22 +272,19 @@ int SqlQuery::prepare(const QByteArray &sql, bool allow_failure)
     if (_stmt) {
         finish();
     }
-    _boundQuery = QString::fromUtf8(_sql);
     if (!_sql.isEmpty()) {
-        int rc = {};
         for (int n = 0; n < SQLITE_REPEAT_COUNT; ++n) {
             qCDebug(lcSql) << "SQL prepare" << _sql << "Try:" << n;
-            rc = sqlite3_prepare_v2(_db, _sql.constData(), -1, &_stmt, nullptr);
-            if (rc != SQLITE_OK) {
+            _errId = sqlite3_prepare_v2(_db, _sql.constData(), -1, &_stmt, nullptr);
+            if (_errId != SQLITE_OK) {
                 qCWarning(lcSql) << "SQL prepare failed" << _sql << QString::fromUtf8(sqlite3_errmsg(_db));
-                if ((rc == SQLITE_BUSY) || (rc == SQLITE_LOCKED)) {
+                if ((_errId == SQLITE_BUSY) || (_errId == SQLITE_LOCKED)) {
```

Comment:
Are you sure you want to write these into the object's state?

Reply:
We do so anyhow a few lines later. And I think we do in the other locations too.
If you are concerend about multithreading mybe we should not set a global state at all.